fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/MetricUnionCompositeTableDefinition.java

```diff
@@ -0,0 +1,111 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.table;
+
+import com.yahoo.bard.webservice.data.config.ResourceDictionaries;
+import com.yahoo.bard.webservice.data.config.dimension.DimensionConfig;
+import com.yahoo.bard.webservice.data.config.names.FieldName;
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.data.time.ZonedTimeGrain;
+import com.yahoo.bard.webservice.metadata.DataSourceMetadataService;
+import com.yahoo.bard.webservice.table.MetricUnionCompositeTable;
+import com.yahoo.bard.webservice.table.PhysicalTable;
+import com.yahoo.bard.webservice.table.PhysicalTableDictionary;
+import com.yahoo.bard.webservice.table.availability.MetricUnionAvailability;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Holds the fields needed to define a Metric Union Composite Table.
+ */
+public class MetricUnionCompositeTableDefinition extends PhysicalTableDefinition {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MetricUnionCompositeTableDefinition.class);
+
+    private final Set<TableName> dependentTableNames;
+
+    /**
+     * Define a physical table using a zoned time grain.
+     *
+     * @param name  The table name
+     * @param timeGrain  The zoned time grain
+     * @param metricNames  The Set of metric names on the table
+     * @param dependentTableNames  The set of dependent table names on the table
+     * @param dimensionConfigs  The dimension configurations
+     */
+    public MetricUnionCompositeTableDefinition(
+            TableName name,
+            ZonedTimeGrain timeGrain,
+            Set<FieldName> metricNames,
+            Set<TableName> dependentTableNames,
+            Set<? extends DimensionConfig> dimensionConfigs
+    ) {
+        super(name, timeGrain, metricNames, dimensionConfigs);
+        this.dependentTableNames = dependentTableNames;
+    }
+
+    /**
+     * Define a physical table with provided logical to physical column name mappings.
+     *
+     * @param name  The table name
+     * @param timeGrain  The zoned time grain
+     * @param metricNames  The Set of metric names on the table
+     * @param dimensionConfigs  The dimension configurations
+     * @param dependentTableNames  The set of dependent table names on the table
+     * @param logicalToPhysicalNames  A map from logical column names to physical column names
+     */
+    public MetricUnionCompositeTableDefinition(
+            TableName name,
+            ZonedTimeGrain timeGrain,
+            Set<FieldName> metricNames,
+            Set<? extends DimensionConfig> dimensionConfigs,
+            Set<TableName> dependentTableNames,
```

Comment:
Let's reorder these two params so that they match the order in the other constructor.